modules/services/etesync-dav.nix

```diff
@@ -0,0 +1,57 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.services.etesync-dav;
+
+in {
+  meta.maintainers = [ maintainers.valodim ];
+
+  options.services.etesync-dav = {
+    enable = mkEnableOption "etesync-dav";
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.etesync-dav;
+      defaultText = "pkgs.etesync-dav";
+      description = "etesync-dav derivation to use.";
+    };
+
+    serverUrl = mkOption {
+      type = types.str;
+      default = "https://api.etesync.com/";
+      description = "The url to the etesync server.";
+    };
+
+    settings = mkOption {
+      type = types.attrsOf types.str;
+      default = { };
+      example = literalExample ''
+        {
+          ETESYNC_LISTEN_ADDRESS = "localhost";
+          ETESYNC_LISTEN_PORT = 37385;
+        }
+      '';
+      description =
+        "Settings for etesync-dav, passed as environment variables.";
+    };
+  };
+
+  config = mkIf cfg.enable (mkMerge [{
+    home.packages = [ cfg.package ];
+
+    systemd.user.services.etesync-dav = {
+      Unit = {
+        Description = "etesync-dav";
+        After = [ "network-online.target" ];
+      };
+
+      Service = { ExecStart = "${cfg.package}/bin/etesync-dav"; };
+
+      environment = { ETESYNC_URL = cfg.serverUrl; } // cfg.settings;
```

Comment:
I'm surprised that this works, I thought systemd expected the environment field to have a specific capitalization and be in the Service section. Do you know where I could find documentation of this?

Reply:
Huh. I have no idea what gave me the impression that this worked, actually. Changed to `Environment=X='Y' A='B'` format